### PR TITLE
[pylintrc] Enforce Google docstring format

### PR DESCRIPTION
**Pull request by SamuelMarks**
Related: #45420 
Requested by: @bhack

### CODE REVIEW

**Comment by kkimdev** [comment]
@mihaimaruseac @av8ramit Do you know if have a similar lint (See #45420) available internally as well?  Linting only for public CI doesn't seem like a good idea.

**Comment by bhack** [comment]
@kkimdev Can you invoke Kokoro with a label in the meantime?

**Comment by bhack** [comment]
@SamuelMarks It Is not working you need one of this solution https://github.com/PyCQA/pylint/issues/735#issuecomment-164144803

**Review by mihaimaruseac — APPROVED** [review]

**Review by av8ramit — APPROVED** [review]

**Review by bhack — CHANGES_REQUESTED** [review]
You need to add `accept-no-param-doc`

**Comment by gbaned** [comment]
@SamuelMarks  Can you please check @bhack's comments and keep us posted ? Thanks!

**Comment by bhack** [comment]
> @SamuelMarks  Can you please check @bhack's comments and keep us posted ? Thanks!

Yes this PR needs to fail tests now and then It need to pass tests after #45420.

**Comment by bhack** [comment]
@gbaned Can you add the kokoro label so that we can check tests?

**Comment by bhack** [comment]
The check Is not working

**Comment by SamuelMarks** [comment]
@bhack If you're talking about the CI, looks like a different issue:
> Executed 2192 out of 2397 tests: 2397 tests pass.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
All tests passed but there were other errors during the build.

Specifically:
> ERROR: /Volumes/BuildData/tmpfs/src/github/tensorflow/tensorflow/lite/python/BUILD:31:1: in deps attribute of py_test rule //tensorflow/lite/python:interpreter_test: '//tensorflow/lite/kernels:deprecated_backends' does not have mandatory providers: 'py' or 'PyInfo'

**Comment by bhack** [comment]
I am talking about that that tests are passing. They need to fail if we need to enforce `arg:`.

**Comment by SamuelMarks** [comment]
🤔
```
$ pylint --rcfile='tensorflow/tools/ci_build/pylintrc' 'tensorflow/compiler/xla/python/xla_client.py'
************* Module python.xla_client
tensorflow/compiler/xla/python/xla_client.py:66:0: W9015: "distributed_client, node_id" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:66:0: W9016: "distributed_client, node_id" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:125:6: R1720: Unnecessary "else" after "raise" (no-else-raise)
tensorflow/compiler/xla/python/xla_client.py:137:0: W9016: "name" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:137:0: W9012: Missing return type documentation (missing-return-type-doc)
tensorflow/compiler/xla/python/xla_client.py:137:0: W9006: "RuntimeError" not documented as being raised (missing-raises-doc)
tensorflow/compiler/xla/python/xla_client.py:153:6: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
tensorflow/compiler/xla/python/xla_client.py:137:0: W9012: Missing return type documentation (missing-return-type-doc)
tensorflow/compiler/xla/python/xla_client.py:159:0: R0205: Class 'OpMetadata' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:159:0: W9015: "op_name, op_type, source_file, source_line" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:159:0: W9016: "op_name, op_type, source_file, source_line" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:170:0: W9015: "op_name, op_type, skip_frames" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:170:0: W9016: "op_name, op_type, skip_frames" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:213:0: W9015: "dtype" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:213:0: W9016: "dtype" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:279:0: W9015: "pyval" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:279:0: W9016: "pyval" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:283:4: R1705: Unnecessary "else" after "return" (no-else-return)
tensorflow/compiler/xla/python/xla_client.py:341:0: W9015: "arguments, backend, executable" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:341:0: W9016: "arguments, backend, executable" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:352:0: W9016: "arguments, backend, executable" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:352:0: W9012: Missing return type documentation (missing-return-type-doc)
tensorflow/compiler/xla/python/xla_client.py:386:0: W9015: "lhs_dims, padding_type, rhs_dims, window_strides" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:386:0: W9016: "lhs_dims, padding_type, rhs_dims, window_strides" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:402:2: R1705: Unnecessary "elif" after "return" (no-else-return)
tensorflow/compiler/xla/python/xla_client.py:426:0: W9016: "fn, name, platform" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:441:0: R0205: Class 'PaddingConfigDimension' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:451:0: R0205: Class 'PaddingConfig' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:472:5: R1701: Consider merging these isinstance calls to isinstance(padding_config, (list, tuple)) (consider-merging-isinstance)
tensorflow/compiler/xla/python/xla_client.py:484:0: R0205: Class 'DotDimensionNumbers' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:512:2: R1705: Unnecessary "else" after "return" (no-else-return)
tensorflow/compiler/xla/python/xla_client.py:524:0: R0205: Class 'ConvolutionDimensionNumbers' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:608:0: R0205: Class 'OpSharding' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:623:0: R0205: Class 'PrecisionConfig' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:633:0: R0205: Class 'GatherDimensionNumbers' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:645:0: R0205: Class 'ScatterDimensionNumbers' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:657:0: R0205: Class 'ReplicaGroup' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tensorflow/compiler/xla/python/xla_client.py:686:0: W9015: "enabled" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:686:0: W9016: "enabled" missing in parameter type documentation (missing-type-doc)
tensorflow/compiler/xla/python/xla_client.py:696:0: W9015: "client" missing in parameter documentation (missing-param-doc)
tensorflow/compiler/xla/python/xla_client.py:33:0: C0411: third party import "from absl import logging" should be placed before "from . import xla_extension as _xla" (wrong-import-order)
tensorflow/compiler/xla/python/xla_client.py:34:0: C0411: third party import "import numpy as np" should be placed before "from . import xla_extension as _xla" (wrong-import-order)

------------------------------------------------------------------
Your code has been rated at 8.47/10 (previous run: 8.47/10, +0.00)
```

**Comment by bhack** [comment]
The lint script that runs in the CI Is https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/ci_sanity.sh

**Comment by bhack** [comment]
Probably we need to extend the pattern at https://github.com/tensorflow/tensorflow/blob/84de9ebadc5d8366ce26980241ebe6e2ca1ad391/tensorflow/tools/ci_build/ci_sanity.sh#L203:L213

**Comment by bhack** [comment]
@mihaimaruseac If we approve #45420 do we want to expand the pattern matching string here over the error list that we care?

/cc @MarkDaoust as it is a doc topic.

File: tensorflow/tools/ci_build/pylintrc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [MASTER]
 
 # Specify a configuration file.
-#rcfile=
+rcfile=tensorflow/tools/ci_build/pylintrc
```
Comment (bhack) [inline tensorflow/tools/ci_build/pylintrc:4]: I don't think you need to change this line

**Review by bhack — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**